src/Shared/RedisUtility.cs

```diff
@@ -75,17 +75,25 @@ internal static object GetObjectFromBytes(byte[] dataAsBytes)
                 return null;
             }
 
-            BinaryFormatter binaryFormatter = new BinaryFormatter();
-            using (MemoryStream memoryStream = new MemoryStream(dataAsBytes, 0, dataAsBytes.Length))
+            try
             {
-                memoryStream.Seek(0, System.IO.SeekOrigin.Begin);
-                object retObject = (object)binaryFormatter.Deserialize(memoryStream);
-
-                if (retObject.GetType() == typeof(RedisNull))
+                BinaryFormatter binaryFormatter = new BinaryFormatter();
```

Comment:
Look at tests in: "aspnet-redis-providers\test\RedisSessionStateProviderUnitTest\RedisUtilityTests.cs" it will give you idea.